src/core/public/overlays/sidecar/sidecar_service.tsx

```diff
@@ -0,0 +1,202 @@
+   * closed this method does nothing.
+   */
+  public close(): Promise<void> {
+    if (!this.closeSubject.closed) {
+      this.closeSubject.next();
+      this.closeSubject.complete();
+    }
+    return this.onClose;
+  }
+}
+
+/**
+ * APIs to open and manage sidecar.
+ *
+ * @public
+ */
+export interface OverlaySidecarStart {
+  /**
+   * Opens a sidecar panel with the given mount point inside. You can use
+   * `close()` on the returned SidecarRef to close the sidecar.
+   *
+   * @param mount {@link MountPoint} - Mounts the children inside a sidecar panel
+   * @param options {@link OverlaySidecarOpenOptions} - options for the sidecar
+   * @return {@link SidecarRef} A reference to the opened sidecar panel.
+   */
+  open(mount: MountPoint, options: OverlaySidecarOpenOptions): SidecarRef;
+
+  /**
+   * Override the default support config
+   * @param config The updated support config
+   */
+  setSidecarConfig(config: Partial<ISidecarConfig>): void;
+
+  /**
+   * Get an observable of the current locked state of the nav drawer.
+   */
+  getSidecarConfig$(): Observable<ISidecarConfig | undefined>;
+
+  /**
+   * Hide side car.
+   */
+  hide: () => void;
+
+  /**
+   * Show side car.
+   */
+  show: () => void;
+}
+
+/**
+ * @public
+ */
+export interface OverlaySidecarOpenOptions {
+  className?: string;
+  'data-test-subj'?: string;
+  config: ISidecarConfig;
+}
+
+interface StartDeps {
+  i18n: I18nStart;
+  targetDomElement: Element | null;
+}
+
+export enum SIDECAR_DOCKED_MODE {
+  LEFT = 'left',
+  RIGHT = 'right',
+  TAKEOVER = 'takeover',
+}
+export interface ISidecarConfig {
+  // takeover mode will docked to bottom
+  dockedMode: SIDECAR_DOCKED_MODE;
+  paddingSize: number;
+  isHidden?: boolean;
+}
+
+/** @internal */
+export class SidecarService {
+  private activeSidecar: SidecarRef | null = null;
+  private targetDomElement: Element | null = null;
+  private readonly stop$ = new ReplaySubject(1);
+
+  public start({ i18n, targetDomElement }: StartDeps): OverlaySidecarStart {
+    this.targetDomElement = targetDomElement;
+    const sidecarConfig$ = new BehaviorSubject<ISidecarConfig | undefined>(undefined);
+    const getSidecarConfig$ = () => sidecarConfig$.pipe(takeUntil(this.stop$));
+
+    const setSidecarConfig = (config: Partial<ISidecarConfig>) => {
+      if (
+        sidecarConfig$.value ||
+        // init
+        (!sidecarConfig$.value && config.dockedMode && config.paddingSize)
+      ) {
+        sidecarConfig$.next({ ...sidecarConfig$.value, ...config } as ISidecarConfig);
+      }
+    };
+
+    return {
+      getSidecarConfig$,
+      setSidecarConfig,
+      open: (mount: MountPoint, options: OverlaySidecarOpenOptions): SidecarRef => {
+        // If there is an active flyout session close it before opening a new one.
+        if (this.activeSidecar) {
+          setSidecarConfig({ paddingSize: 0 });
+          this.activeSidecar.close();
```

Comment:
This line will trigger the promise in line 156 I suppose, call `cleanupDom` one time in the promise should be enough?